lib/hammer_cli_foreman_ansible/command_extensions/resolver.rb

```diff
@@ -1,11 +1,7 @@
 module HammerCLIForemanAnsible
   module ResolverExtension
     def create_ansible_roles_search_options(options, mode = nil)
-      if defined? HammerCLIKatello::IdResolver
```

Comment:
The easiest fix that comes to my mind is

```ruby
if self.class.to_s == 'HammerCLIKatello::IdResolver'
  create_search_options_without_katello_api(options, api.resource(:ansible_roles), mode)
else
...
```

This should ensure that this method will be used if the current resolver comes from katello plugin. `to_s` is needed in case if katello plugin is not installed, so we don't crash with `undefined constant HammerCLIKatello`.

But I agree that this is still error-prone, so we should find a better way to deal with IdResolvers. I can imagine that different plugins will try to redefine a command to use its own resolver which will lead to similar problems...